tests/test_tokenizers.py

```diff
@@ -0,0 +1,37 @@
+import pytest
+
+from guidance import models
+
+
+class TestTransformerTokenizers:
+    @pytest.mark.parametrize(
+        "model_name",
+        [
+            "gpt2",
+            "microsoft/Phi-3-mini-4k-instruct",
+            "microsoft/Phi-3-vision-128k-instruct",
+            "microsoft/phi-2",
+        ],
+    )
+    def test_smoke(self, model_name: str):
+        my_tok = models.TransformersTokenizer(model=model_name, tokenizer=None)
+        assert my_tok is not None
+
+    @pytest.mark.parametrize(
+        "model_name",
+        [
+            "gpt2",
+            "microsoft/Phi-3-mini-4k-instruct",
+            "microsoft/Phi-3-vision-128k-instruct",
+            "microsoft/phi-2",
+        ],
+    )
+    @pytest.mark.parametrize("target_string", ["hello", "’"])
+    def test_string_roundtrip(self, model_name: str, target_string: str):
+        my_tok = models.TransformersTokenizer(model=model_name, tokenizer=None)
+
+        encoded = my_tok.encode(target_string.encode())
+        decoded = my_tok.decode(encoded)
+        final_string = decoded.decode()
+
+        assert final_string == target_string
```

Comment:
Really like the way this is looking, but we should significantly extend the string space on the parametrize to include model-specific special tokens where encode/decode cycles are most likely to have failures (e.g. around role tags in chat formats)

Reply:
I felt that a single failing test was enough to get started

Reply:
Yes of course! We should do the same with Llamacpp tokenizers too (our other local tokenization pathway)